libs/langchain/langchain/document_loaders/pdf.py

```diff
@@ -440,3 +476,127 @@ def load(self) -> List[Document]:
+    the name of the profile from the ~/.aws/credentials file that is to be used.
+
+    Make sure the credentials / roles used have the required policies to
+    access the Amazon Textract service.
+
+    Example:
+        .. code-block:: python
+            from langchain.document_loaders import AmazonTextractPDFLoader
+            loader = AmazonTextractPDFLoader(
+                file_path="s3://pdfs/myfile.pdf"
+            )
+            document = loader.load()
+    """
+
+    try:
+        import textractcaller as tc
+    except ImportError:
+        raise ModuleNotFoundError(
+            "Could not import amazon-textract-caller python package. \
+            Please install it with `pip install amazon-textract-caller` \
+            or add to your requirements.txt."
+        )
+
+    def __init__(
+        self,
+        file_path: str,
+        textract_features: List[tc.Textract_Features] = list(),
+        client: Any = None,
+        credentials_profile_name: str = "",
+        region_name: str = "",
+        endpoint_url: str = "",
+    ) -> None:
+        super().__init__(file_path)
+        if credentials_profile_name or region_name or endpoint_url:
+            try:
+                import boto3
+
+                if credentials_profile_name is not None:
+                    session = boto3.Session(profile_name=credentials_profile_name)
+                else:
+                    # use default credentials
+                    session = boto3.Session()
+
+                client_params = {}
+                if region_name:
+                    client_params["region_name"] = region_name
+                if endpoint_url:
+                    client_params["endpoint_url"] = endpoint_url
+
+                client = session.client("textract", **client_params)
+
+            except ImportError:
+                raise ModuleNotFoundError(
+                    "Could not import boto3 python package. "
+                    "Please install it with `pip install boto3`."
+                )
+            except Exception as e:
+                raise ValueError(
+                    "Could not load credentials to authenticate with AWS client. "
+                    "Please check that credentials in the specified "
+                    "profile name are valid."
+                ) from e
+        self.parser = AmazonTextractPDFParser(
+            textract_features=textract_features, client=client
+        )
+
+    def load(self) -> List[Document]:
+        """Load given path as pages."""
+        return list(self.lazy_load())
+
+    def lazy_load(
+        self,
+    ) -> Iterator[Document]:
+        """Lazy load documents"""
+        # the self.file_path is local, but the blob has to include
+        # the S3 location if the file originated from S3 for multi-page documents
+        # raises ValueError when multi-page and not on S3"""
+        blob = Blob.from_path(self.file_path)
+        if AmazonTextractPDFLoader._get_number_of_pages(blob) > 1:
```

Comment:
we probably want to flip the order of the if statements, first check if it's an s3 url, and then if not, check the number of pages